src/ProjectTemplates/test/Helpers/AspNetProcess.cs

```diff
@@ -83,6 +85,59 @@ public void VisitInBrowser(IWebDriver driver)
             }
         }
 
+        public async Task AssertPagesOk(IEnumerable<string> pages, IEnumerable<string> expectedLinks)
+        {
+            foreach (var page in pages)
+            {
+                await AssertOk(page);
+                await ContainsLinks(page, expectedLinks);
+                await ExternalLinksWork(page);
+            }
+        }
+
+        public async Task ContainsLinks(string requestUrl, IEnumerable<string> expectedLinks)
```

Comment:
I'm not sure this is covering things like `<link rel="stylesheet" type="text/css" href="theme.css">` and similar. Or `<script src="app.js" />` or for that matter. Are we not planning to cover those?

Also, shouldn't we fail if we encounter more links than expected? That would mean that we added something we are not testing.

Reply:
This is definitely not covering thinks like missing style-sheets/scripts. My thought was to cover that using the Browser console checks we have to add anyway for the LibMan switch, but that would mean testing non-SPA templates with Selenium. May as well start doing it here since we're already parsing.